internal/blocksync/pool.go

```diff
@@ -154,25 +147,16 @@ func (pool *BlockPool) removeTimedoutPeers() {
 	}
 }
 
-// GetStatus returns pool's height, numPending requests and the number of
-// requesters.
-func (pool *BlockPool) GetStatus() (height int64, numPending int32, lenRequesters int) {
-	pool.mtx.Lock()
-	defer pool.mtx.Unlock()
-
-	return pool.height, atomic.LoadInt32(&pool.numPending), len(pool.requesters)
-}
-
 // IsCaughtUp returns true if this node is caught up, false - otherwise.
 // TODO: relax conditions, prevent abuse.
-func (pool *BlockPool) IsCaughtUp() bool {
+func (pool *BlockPool) IsCaughtUp() (isCaughtUp bool, height, maxPeerHeight int64) {
```

Comment:
`height` and `maxPeerHeight` are only returned for logging messages. An alternative is to change their types from `int64` to `atomic.Int64` to access them safely when needed.

Reply:
I think this is a snapshot from the pool state. Namely, one should expect the information to sometimes be outdated. I don't see much point of creating another point of contention in order to return an atomic version of these counters.